src/renderer/src/lib.rs

```diff
@@ -99,6 +101,60 @@ impl<R, C> Renderer<R, C>
     }
 }
 
+/// holds a 1x1 texture that can be used to store constnat colours
+pub struct ConstantColorTexture<R: gfx::Resources> {
+    texture: gfx::handle::Texture<R, gfx::format::R8_G8_B8_A8>,
+    view: gfx::handle::ShaderResourceView<R, [f32; 4]>
+}
+
+impl<R: gfx::Resources> ConstantColorTexture<R> {
+    /// Create a texture buffer
+    pub fn new<F>(factory: &mut F) -> ConstantColorTexture<R>
+        where F: gfx::Factory<R>
+    {
+        let kind = gfx::tex::Kind::D2(1, 1, gfx::tex::AaMode::Single);
+        let (text, view) = factory.create_texture_const_u8::<gfx::format::Rgba8>(kind, &[&[0, 0, 0, 0]]).unwrap();
```

Comment:
You can't use `create_texture_const` here, it's only for immutable textures. You'd need to explicitly say that this is a dynamic texture by creating the texture object manually with `create_texture` and passing `Usage::Dynamic` in there.